packages/sit-onyx/src/composables/useClipping.ts

```diff
@@ -0,0 +1,92 @@
+import { useGlobalEventListener } from "@sit-onyx/headless";
+import { computed, nextTick, onMounted, ref, type Ref } from "vue";
+
+export function useClipping(
+  popoverRef: Ref<HTMLElement | null>,
+  popoverWrapperRef: Ref<HTMLElement | null>,
+  popoverPosition: Ref<string>,
+  isVisible: Ref<boolean>,
+  clipping: Ref<boolean>,
+) {
+  const clippingStyles = ref("");
+  const scrolledOut = ref<null | "top" | "bottom">(null);
+  const isClipping = ref(false);
+
+  const checkVisibilityOnScroll = () => {
+    const MIN_DISTANCE_TO_BORDER = 16;
+    const MARGIN = 8;
+
+    if (!popoverRef.value || !popoverWrapperRef.value || !isVisible.value) return;
+
+    if (isClipping.value) {
+      const popoverRect = popoverRef.value.getBoundingClientRect();
+      const wrapperRect = popoverWrapperRef.value.getBoundingClientRect();
+      const requiredHeight = popoverRect.height + MIN_DISTANCE_TO_BORDER + MARGIN;
+
+      if (
+        popoverPosition.value === "top" ||
+        popoverPosition.value === "top left" ||
+        popoverPosition.value === "top right"
```

Comment:
```suggestion
        popoverPosition.value.includes("top")
```

Shouldn't this be the same?